### ISSUE TITLE
Change papis configuration file format into yaml

### ISSUE THREAD

**Issue by alejandrogallo** [papis/papis#282]
I propose changing the configuration
file format of papis from the `INI` file format
into `yaml`.

This would simplify working in the code
with the configuration and we would have a yaml
file in the codebase with all available configuration
options plus their respective documentation.

**Comment by alejandrogallo** [comment]
Working on this in a separate project: https://github.com/alejandrogallo/konfigurazioa

**Comment by jorsn** [comment]
I looked at the requirements for *konfigurazioa*:

- [X] Basic types
- [X] Optional types
- [X] Lists (typed or untyped)
- [X] Union types
- [ ] Callables (lambdas etc)
- [ ] Python expressions
- [ ] Guards
- [ ] User defined types

[Dhall](https://dhall-lang.org/) supports everything except python expressions and untyped terms.

Python expressions can still be embedded as strings (just like bash code is embedded in [nix expressions](https://nixos.org) as strings for building packages). Of course this is not particularly safe.

You would need an interface to Dhall. This could be either JSON/YAML/XML or you have a look at [python-dhall](https://github.com/TristanCacqueray/python-dhall).

**Comment by alejandrogallo** [comment]
Yeah I love dhall and if `python-dhall` weren't so young and not portable to
windows (maybe I'm wrong, I'd be glad to hear from it) I'd use it, but right
now there is nothing I've found that really fills the gap in my opinion, that's
why I wanted a small but very functional solution for this, and `konfigurazioa`
should just be a single readable document defining some simple structures that
one can use. I agree about the security of pythion expressions, but then again
is like people will use papis configurations that they haven't read before. 

About dhall, for instance I'd plan to add a hook that
before the configuration is read some command is called, then you
could use dhall to yaml converter for this.
Actually I kind of plan to do something like this to define
papis libraries in the wild, so that you cann add them from github or whatever
if the'yre to be read only.

Konfigurazioa is almost ready, just have to finish off some tests and
write the document so that it is understandable to everyone.
However time is not on my side... :(
